multicast/discovery.go

```diff
@@ -134,6 +137,25 @@ func (d *Discovery) Respond(ctx context.Context, info ServerInfo) error {
 	return nil
 }
 
+// StopResponder stops the responder server and cancels it's inner context.
+func (d *Discovery) StopResponder() error {
+	// Check if this instance of discovery has an active responder server connection.
+	if d.responderConn != nil {
+		err := d.responderConn.Close()
+		// Ignore errors if the connection is already closed.
+		// This can happen if the responders context already got cancelled
+		// which also triggers a close of the connection.
+		if err != nil && !errors.Is(err, net.ErrClosed) {
```

Comment:
`net.ErrClosed` is the sentinel error for closed network connections, but depending on how the error is wrapped, is there a chance that `errors.Is` may not correctly identify it and lead to unexpected error handling?

Reply:
If we call `StopResponder` there can be two situations:

1) The responder is still running (connection still alive and running go routines)
2) The responder is already stopped (connection closed and context cancelled)

Now if we try to close the connection but it is already closed due to the context being cancelled before (we cannot control this as its the outer context), it would be annoying to see an error from closing the connection. Therefore this one is being filtered out. 

If it's wrapped differently we will return it as there is clearly something else going on and we would like to see it.